librz/search/string_search.c

```diff
@@ -0,0 +1,149 @@
+// SPDX-FileCopyrightText: 2024 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2024 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_search.h>
+#include <rz_util.h>
+
```

Comment:
Please add `\file` to note that it searches not only ANSI/ASCII but also Unicode and some other encodings